internal/filter/filterlog/filterlog.go

```diff
@@ -15,23 +15,38 @@
 package filterlog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterlog"
 
 import (
+	"context"
 	"fmt"
 
 	"go.opentelemetry.io/collector/pdata/pcommon"
-	"go.opentelemetry.io/collector/pdata/plog"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterconfig"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filtermatcher"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterset"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottllog"
 )
 
-// Matcher is an interface that allows matching a log record against a
-// configuration of a match.
-// TODO: Modify Matcher to invoke both the include and exclude properties so
-//
-//	calling processors will always have the same logic.
-type Matcher interface {
-	MatchLogRecord(lr plog.LogRecord, resource pcommon.Resource, library pcommon.InstrumentationScope) bool
+// NewSkipExpr creates a BoolExpr that on evaluation returns true if a log should NOT be processed or kept.
+// The logic determining if a span should be processed based on include and exclude settings.
+// Include properties are checked before exclude settings are checked.
+func NewSkipExpr(mp *filterconfig.MatchConfig) (expr.BoolExpr[ottllog.TransformContext], error) {
+	var matchers []expr.BoolExpr[ottllog.TransformContext]
+	inclExpr, err := newExpr(mp.Include)
```

Comment:
Since `include` settings will always be negated is there any value in a `newIncludeExpr` that knows how to do the negating?  It would be great if the negating was built into the concept of `mp.Include` so that we don't have to know to negate it ourselves when creating a new `SkipExpr`.

Reply:
I can followup on this. For the moment this is good enough.